README.md

```diff
@@ -208,181 +181,26 @@ The has_ancestry method supports the following options:
-example:
-
-```ruby
-node.children.where(:name => 'Mary').exists?
-node.subtree.order(:name).limit(10).each { ... }
-node.descendants.count
-```
-
-For convenience, a couple of named scopes are included at the class level:
-
-    roots                   Root nodes
-    ancestors_of(node)      Ancestors of node, node can be either a record or an id
-    children_of(node)       Children of node, node can be either a record or an id
-    descendants_of(node)    Descendants of node, node can be either a record or an id
-    indirects_of(node)      Indirect children of node, node can be either a record or an id
-    subtree_of(node)        Subtree of node, node can be either a record or an id
-    siblings_of(node)       Siblings of node, node can be either a record or an id
-
-Thanks to some convenient rails magic, it is even possible to create nodes
-through the children and siblings scopes:
-
-    node.children.create
-    node.siblings.create!
-    TestNode.children_of(node_id).new
-    TestNode.siblings_of(node_id).create
-
-# Selecting nodes by depth
-
-When depth caching is enabled (see has_ancestry options), five more named
-scopes can be used to select nodes on their depth:
-
-    before_depth(depth)     Return nodes that are less deep than depth (node.depth < depth)
-    to_depth(depth)         Return nodes up to a certain depth (node.depth <= depth)
-    at_depth(depth)         Return nodes that are at depth (node.depth == depth)
-    from_depth(depth)       Return nodes starting from a certain depth (node.depth >= depth)
-    after_depth(depth)      Return nodes that are deeper than depth (node.depth > depth)
-
-The depth scopes are also available through calls to descendants,
-descendant_ids, subtree, subtree_ids, path and ancestors. In this case, depth
-values are interpreted relatively. Some examples:
-
-    node.subtree(:to_depth => 2)      Subtree of node, to a depth of node.depth + 2 (self, children and grandchildren)
-    node.subtree.to_depth(5)          Subtree of node to an absolute depth of 5
-    node.descendants(:at_depth => 2)  Descendant of node, at depth node.depth + 2 (grandchildren)
-    node.descendants.at_depth(10)     Descendants of node at an absolute depth of 10
-    node.ancestors.to_depth(3)        The oldest 4 ancestors of node (its root and 3 more)
-    node.path(:from_depth => -2)      The node's grandparent, parent and the node itself
-
-    node.ancestors(:from_depth => -6, :to_depth => -4)
-    node.path.from_depth(3).to_depth(4)
-    node.descendants(:from_depth => 2, :to_depth => 4)
-    node.subtree.from_depth(10).to_depth(12)
-
-Please note that depth constraints cannot be passed to ancestor_ids and
-path_ids. The reason for this is that both these relations can be fetched
-directly from the ancestry column without performing a database query. It
-would require an entirely different method of applying the depth constraints
-which isn't worth the effort of implementing. You can use
```

Comment:
these remind me of offset and limit
or slice

would be great if we implemented methods that work on associations
not sure if it is worth the non-backwards compatibility or implementing both